src/hooks/useUploadingFilesData/useUploadingFilesData.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import {
+  getUploadingFiles,
+  getUploadingFile as getUploadingFileFn,
+  updateUploadingFileStatus as updateUploadingFileStatusFn,
+  addUploadingFileData as addUploadingFileDataFn,
+  removeUploadingFileData as removeUploadingFileDataFn,
+  clearUploadingFilesData as clearUploadingFilesDataFn,
+} from './utils'
+import { CropData } from '@/components'
+
+type ParentObjectType = 'video' | 'channel'
+type UploadingFileType = 'video' | 'thumbnail' | 'cover' | 'avatar'
+export type StatusType = 'completed' | 'notCompleted'
+export type UploadingFile = {
```

Comment:
We will need few more fields here:
- some info on when this was scheduled for upload, maybe `updatedAt`?
- `size`
- we'll probably also want some metadata, for example `1920x1080`. for now maybe let's just add `metadata?: string`